client/src/lib/dependency-injection/buildReducerContainer.js

```diff
@@ -1,7 +1,69 @@
 import buildBaseContainer from './buildBaseContainer';
+import MiddlewareRegistry from './MiddlewareRegistry';
 
 const buildReducerContainer = (base = buildBaseContainer()) => ({
   ...base,
+
+  /**
+   * The redux store which will be providing the global context for the reducer middleware
+   * @type {object}
+   */
+  store: null,
+
+  setStore(store) {
+    if (this.initialised) {
+      throw new Error('Cannot mutate DI container after it has been initialised');
+    }
```

Comment:
It got impractical to have the `protect()` wrapper for the methods, which then needed rebinding to that object.
I guess it could have a `base.protect.call(this);` helper method if desired.